sel4bench/build.py

```diff
@@ -126,6 +133,203 @@ def make_runs(builds: List[Build]) -> List[Run]:
+                    ipc_reply = round(result['Mean'])
+                    ipc_reply_s = round(result['Stddev'])
+        if bench['Benchmark'].startswith('IRQ path cycle count'):
+            results = bench['Results']
+            for result in results:
+                if result['Type'] == 'Without context switch':
+                    irq_invoke = round(result['Mean'])
+                    irq_invoke_s = round(result['Stddev'])
+
+        if ipc_call > 0 and ipc_reply > 0 and irq_invoke > 0:
+            break
+
+    return [irq_invoke, irq_invoke_s, ipc_call, ipc_call_s, ipc_reply, ipc_reply_s]
+
+
+def get_run(runs: List[Run], name: str) -> Optional[Run]:
+    """Get a run by name."""
+
+    for run in runs:
+        if run.name == name:
+            return run
+
+    print(f"Run {name} not found.")
+    return None
+
+
+def gen_web(runs: List[Run], yml, file_name: str):
+    """Generate web page for benchmark results according to the set defined in builds.yml"""
+
+    manifest_sha = os.getenv('INPUT_MANIFEST_SHA')
+
+    boards = yml["boards"]
+    sections = yml["results"]
+
+    th_style = {
+        0: 'style=""',
+        1: 'style="text-align: center;"',
+        2: 'style="text-align: center;"'
+    }
+
+    # (column name, th_style, span, width)
+    cols = [
+        ('ISA', 0, 1, "8ex"),
+        ('Mode', 2, 1, "4ex"),
+        ('Core/SoC/Board', 0, 1, "40%"),
+        ('Clock', 1, 1, "8ex"),
+        ('IRQ Invoke', 1, 2, ""),
+        ('IPC call', 1, 2, ""),
+        ('IPC reply', 1, 2, "")
+    ]
+
+    with open(file_name, 'w') as f:
+        f.write('<!-- <title>seL4 benchmarks</title> -->\n')
+        f.write('<!--\n')
+        f.write('Copyright 2021 seL4 Project a Series of LF Projects, LLC.\n')
+        f.write('SPDX-License-Identifier: CC-BY-SA-4.0\n')
+        f.write('-->\n\n')
+
+        f.write('<h1>Performance</h1>\n')
+        f.write('<p>This page displays the latest benchmark numbers for seL4 from the publicly\n')
+        f.write('available <a href="https://github.com/seL4/sel4bench">sel4bench repository</a>.\n')
+        f.write('The following times are reported as mean and standard deviation in\n')
+        f.write('the format <em>mean (std dev)</em>, both rounded to the nearest integer.</p>\n')
+
+        f.write('<ul>')
+        f.write('<li><strong>IRQ invoke</strong>: ')
+        f.write('Time in cycles to invoke a user-level interrupt handler running in the same\n')
+        f.write('address space as the interrupted thread.</li>\n')
+
+        f.write('<li><strong>IPC call</strong>: ')
+        f.write('Time in cycles for invoking a server in a different address space on the same core.</li>\n')
+
+        f.write('<li><strong>IPC reply</strong>: ')
+        f.write('Time in cycles for a server replying to a client in a different address space on\n')
+        f.write('the same core.</li>\n')
+        f.write('</ul>')
```

Comment:
could use triple-quote blocks to reduce the `f.write(` calls

Reply:
The problem with the triple quotes is that the indent then looks ugly. We could use string concatenation to write a bit less, but I'm not sure it adds much.